embed-code-go/fragmentation/fragment_file.go

```diff
@@ -0,0 +1,130 @@
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+package fragmentation
+
+import (
+	"crypto/sha1"
+	"embed-code/embed-code-go/configuration"
+	"encoding/hex"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// A file storing a single fragment from the file.
+type FragmentFile struct {
+	CodeFile      string                      // a relative path to a code file
+	FragmentName  string                      // a name of the fragment in the code file
+	Configuration configuration.Configuration // a configuration for embedding
+}
+
+// Iniitalizers
+
+// Composes a FragmentFile for the given fragment in the given code file.
+//
+// codeFile is an absolute path to a code file.
+//
+// fragmentName is a name of the fragment in the code file.
+//
+// configuration is a configuration for embedding.
+//
+// Returns composed fragment.
+func NewFragmentFileFromAbsolute(
+	codeFile string,
+	fragmentName string,
+	configuration configuration.Configuration,
+) FragmentFile {
+
+	absoluteCodeRoot, err := filepath.Abs(configuration.CodeRoot)
+	if err != nil {
+		panic(err)
+	}
+	relativeCodeFile, err := filepath.Rel(absoluteCodeRoot, codeFile)
+	if err != nil {
+		panic(err)
+	}
+
+	return FragmentFile{
+		CodeFile:      relativeCodeFile,
+		FragmentName:  fragmentName,
+		Configuration: configuration,
+	}
+}
+
+//
+// Public methods
+//
+
+// Writes text to the file.
+// Overwrites the file if it exists.
+func (fragmentFile FragmentFile) Write(text string) {
+	byteStr := []byte(text)
+	filePath := fragmentFile.absolutePath()
+	os.WriteFile(filePath, byteStr, 0777)
+}
+
+// Reads content of the file.
+//
+// Return contents of the file as a list of strings or raises an error if it doesn't exists.
+func (fragmentFile FragmentFile) Content() []string {
+	path := fragmentFile.absolutePath()
+	isPathFileExits, err := IsFileExists(path)
+	if isPathFileExits {
+		return ReadLines(path)
+	} else {
+		panic(err)
+	}
+}
+
+func (fragmentFile FragmentFile) String() string {
+	return fragmentFile.absolutePath()
+}
+
+//
+// Private methods
+//
+
+// Obtains the absolute path to this fragment file.
+func (fragmentFile FragmentFile) absolutePath() string {
+
+	fileExtension := filepath.Ext(fragmentFile.CodeFile)
+	fragmentsAbsDir, err := filepath.Abs(fragmentFile.Configuration.FragmentsDir)
+	if err != nil {
+		panic(err)
+	}
+
+	if fragmentFile.FragmentName == DefaultFragment {
+		return filepath.Join(fragmentsAbsDir, fragmentFile.CodeFile)
+	} else {
+		withoutExtension := strings.TrimSuffix(fragmentFile.CodeFile, fileExtension)
+		filename := fmt.Sprintf("%s-%s", withoutExtension, fragmentFile.getFragmentHash())
+		return filepath.Join(fragmentsAbsDir, filename+fileExtension)
+	}
+}
+
+// Creates and returns a hash string for FragmentFile.
+// Since fragments which have the same name unite into one
+// fragment with multiple partitions, the name of a fragment is unique.
+func (fragmentFile FragmentFile) getFragmentHash() string {
```

Comment:
Please refrain from naming methods with `get`. This is a bad practice, especially for those methods which do not plain-return values, but _calculate_ them, as it is in this case.